packages/data-store/src/contact/ContactStore.ts

```diff
@@ -123,13 +123,58 @@ export class ContactStore extends AbstractContactStore {
   }
 
   removeContact = async ({ contactId }: IRemoveContactArgs): Promise<void> => {
-    debug('Removing contact', contactId)
-    ;(await this.dbConnection)
+    debug('Removing contact', contactId);
+
+    (await this.dbConnection)
       .getRepository(ContactEntity)
-      .delete({ id: contactId })
+      .findOneById(contactId)
+      .then(async (contact: ContactEntity | null) => {
+        if (!contact) {
+          await Promise.reject(Error(`Unable to find the contact with id to remove: ${contactId}`))
+        } else {
+          await this.deleteIdentities(contact.identities)
+
+          await (await this.dbConnection)
+            .getRepository(ContactEntity)
+            .delete({ id: contactId })
+            .catch((error) => Promise.reject(Error(`Unable to remove contact with id: ${contactId}. ${error}`)))
+        }
+      })
       .catch((error) => Promise.reject(Error(`Unable to remove contact with id: ${contactId}. ${error}`)))
   }
 
+  private async deleteIdentities(identities: Array<IdentityEntity>): Promise<void> {
+    debug('Removing identities', identities)
+
+    identities.map(async (identity: IdentityEntity) => {
+      await (await this.dbConnection)
+        .getRepository(CorrelationIdentifierEntity)
+        .delete(identity.identifier.id)
+        .catch((error) => Promise.reject(Error(`Unable to remove identity.identifier with id: ${identity.identifier.id}. ${error}`)))
+
+      if (identity.connection) {
```

Comment:
are you sure that the config relation is deleted with this?

Reply:
No, it was not deleted earlier. Now it will be.